### PR TITLE
Feature/6143 user saved products

### PR DESCRIPTION
**Pull request by s3341458**
It is not my will, but I did done some 'architectural level of changes' due to the fact I was trapped in dependency problem.

@DanielCollins please have a look and give your opinions.

### CODE REVIEW

**Comment by DanielCollins** [comment]
what's the real problem that needs to be solved?

**Comment by s3341458** [comment]
try to add
"from sdk.python.products import Product"
to python/companies.py
then you will see the problem that I was experiencing.

**Comment by s3341458** [comment]
oh if you mean the targeting problem. the following two was the initial goal that I want to achieve.
in python/companies.py:
>    # products that are saved by company for future reference
>    saved_products = Property(sdk.python.products.Product,
                              backref="saved_by_companies")
in python/users.py
>    # products that are saved by user for future reference
>    saved_products = Property(Product, backref="saved_by_users")

File: python/entities.py [hunk 1]
```diff
@@ -42,16 +44,25 @@ def wrap_init(*args, **kwargs):
...
-            if isinstance(value, Property) and getattr(value.remote_type,
-                                                       '_is_merchi_entity',
-                                                       False) \
-               and value.backref:
-
+            if isinstance(value, Property) and value.backref:
+                # TODO: 'Dynamic module load should also support backref'
+                if isinstance(value.remote_type, str):
+                    raise NotImplementedError('Backref has not been supported'
+                                              'for dynamic module reference')
```
Comment (DanielCollins) [inline python/entities.py]: well, better fix this up

Comment (s3341458) [reply python/entities.py]: yeah please let me do that in a separate branch, since I reckon it is "quite difficult to me"

Comment (DanielCollins) [reply python/entities.py]: no TODO stuff in master please. if the branch isn't ready it isn't ready.

File: python/entities.py [hunk 2]
```diff
@@ -15,6 +17,13 @@
 from jinja2 import utils
 
 
+backref_globals = {}  # type: Dict[Any, Any]
+
+
+def full_class_path(obj):
```
Comment (DanielCollins) [inline python/entities.py]: missing a docstring here

**Pull request merged**